src/state/reducers/scigateway.reducer.tsx

```diff
@@ -92,12 +94,62 @@ export function handleConfigureStrings(
   };
 }
 
+const updatePluginSrc = (
+  existingPlugins: PluginConfig[],
+  payload: RegisterPluginSrcPayload[]
+): PluginConfig[] => {
+  payload.forEach((plugin) => {
+    let match = false;
+    // If we already have the plugin, add the src url
+    existingPlugins.forEach((existingPlugin) => {
+      if (existingPlugin.plugin === plugin.name) {
+        existingPlugin.src = plugin.src.split('/main.js')[0];
+        match = true;
+      }
+    });
+    // If we don't, create a placeholder with the src
+    if (!match) {
+      const newPlugin = {
+        plugin: plugin.name,
+        src: plugin.src.split('/main.js')[0],
+        section: '',
+        link: '',
+        displayName: '',
+        order: -1,
+      };
+      existingPlugins.push(newPlugin);
+    }
+  });
+  return existingPlugins;
+};
```

Comment:
Why were these changes needed? Was there a problem of duplicate plugins or something? I can't fathom why these changes are needed for the logo change unless it's fixing a separate problem.

Reply:
The problem I was having was the `src` string for the image that the DG plugin registers (`plugin.logoDarkMode`/`plugin.logoLightMode`) doesn't have the full address, it's something like `/static/media/datagateway-logo.808b0870.svg` but to display the image in Scigateway I needed the `http://localhost:5001` bit as well. The only place I could find to get this was when we read SG's `settings.json` file, and then add the address to the plugin so when we display the image we can combine the two.

I wasn't sure if we registered the plugins or read the settings first, so I put in logic to handle it either way round. If the plugin isn't registered it created a placeholder with just the name and `http://localhost:5001` or whatever, then we actually register the plugin later it adds the rest of the information to that entry in the list of plugins. Or if the plugin is already registered, we just set the `src` field on the existing entry.

Like I said this was the only way I could find of getting the full address needed for hosting the image. If there's a better way of getting this to work I'd be more than happy to change it; I agree that it's not the nicest thing to look at.